src/datasources/cache/cache.router.ts

```diff
@@ -48,6 +49,20 @@ export class CacheRouter {
     );
   }
 
+  static getValkBalancesCacheKey(args: {
+    chainId: string;
+    safeAddress: string;
+  }): string {
+    return `${args.chainId}_${CacheRouter.VALK_BALANCES_KEY}_${args.safeAddress}`;
+  }
+
+  static getValkBalancesCacheDir(args: {
+    chainId: string;
+    safeAddress: string;
+  }): CacheDir {
+    return new CacheDir(CacheRouter.getValkBalancesCacheKey(args), '');
+  }
```

Comment:
I think having separated cache prefixes by provider is the way to go. Otherwise, the balances would be cached regardless of the provider, so changing the configuration of a chain could make an integration process the response of a different provider and therefore erroring for some time (until it expired). 

Once the migration is complete and only one provider is used, the provider prefix can be removed.

Reply:
Let's add a TODO to remove the prefix.

Reply:
Added in [4f5723f](https://github.com/safe-global/safe-client-gateway/pull/1062/commits/4f5723f06f49a21b090547d02f90155e3a297cd2)